webapp/src/pages/_app.tsx

```diff
@@ -20,8 +20,18 @@ type AppPropsWithLayout = AppProps & {
   Component: NextPageWithLayout;
 };
 
-const DogmaApp = ({ Component, pageProps }: AppPropsWithLayout) => {
+const DogmaApp = ({Component, pageProps}: AppPropsWithLayout) => {
   const router = useRouter();
+  useEffect(() => {
+    // Path patterns are used folder structures in Next.js such as '/app/projects/[projectNames]'.
+    // If '/app/projects/myProj' is requested to Central Dogma server, the server fails to find
+    // 'app/projects/myProj/index.html' file and returns 'index.html' as a fallback and the landing page is
+    // rendered instead. As a workaround, this triggers Next.js router to route to the desired page when a page
+    // is loaded for the first time.
```

Comment:
> is loaded for the first time.

It seems like `router.asPath` is always pushed every time a state change is made. I'm wondering if this is the intended behavior?

Reply:
This should be called only once when the web application is initialized.
Is this repeatedly called?

Reply:
I'm not sure, but aren't effects called every time a state is changed?

Reply:
`[]` is set to the dependency of `useEffect()` so it should be run once.
https://css-tricks.com/run-useeffect-only-once/#aa-the-trick-is-that-useeffect-takes-a-second-parameter